src/test/java/org/apache/sling/scripting/sightly/compiler/expression/nodes/BinaryOperatorTest.java

```diff
@@ -106,4 +106,44 @@ public void testStrictEqWrongType() {
 
     }
 
+    @RunWith(Parameterized.class)
+    public static class NumberError {
+
+        @Parameters(name = "Comparison: {2}")
+        public static Iterable<? extends Object> data() {
+            List<Object[]> list = new ArrayList<>();
+            list.add(new Object[]{1, new Object(), "number to object not equal"});
+            list.add(new Object[]{new Object(), 2, "object to number not equal"});
+            list.add(new Object[]{TestEnum.ONE, 1, "String to number not equal"});
```

Comment:
Here I think the correct message is "enum to number not equal". You would need a different test for comparing a String to a number.

Reply:
@raducotescu: I have added test coverage for `string to number` and corrected the relevant messages